### PR TITLE
Replace on_trait_change with observe

### PR DESCRIPTION
**Pull request by aaronayres35**
**Checklist**
- [x] Add a news fragment if this PR is news-worthy for end users. (see docs/releases/README.rst)

This PR replaces _almost_ all uses of `on_trait_change` with observe.

All that remains is `apptools.selection.selection_service.py` and `apptools.scripting.recorder.py` (also an undo example but that is deprecated here anyway).  These are a bit trickier and can be addressed separately

### CODE REVIEW

File: apptools/preferences/ui/widget_editor.py [hunk 1]
```diff
@@ -58,14 +58,14 @@ def update_editor(self):
     # Private interface.
     ###########################################################################
 
-    def _on_page_changed(self, obj, trait_name, old, new):
+    def _on_page_changed(self, event):
```
Comment (rahulporuri) [inline apptools/preferences/ui/widget_editor.py:61]: Maybe rename to `_handle_change_to_selected_page`?

Also, i'm not sure why we're not using the decorator here,

Comment (rahulporuri) [reply apptools/preferences/ui/widget_editor.py:61]: It's quite possible that this code was written before the decorator existed in traits - i don't see a good reason why we can't use the new decorator here.

**Review by rahulporuri — APPROVED** [review]
LGTM with a few comments/suggestions.

File: apptools/preferences/preference_binding.py [hunk 2]
```diff
@@ -121,7 +121,7 @@ def _initialize(self):
         """ Wire-up trait change handlers etc. """
 
         # Listen for the object's trait being changed.
-        self.obj.on_trait_change(self._on_trait_changed, self.trait_name)
+        self.obj.observe(self._on_trait_changed, self.trait_name)
```
[the remark this answers is not in the document]
Comment (aaronayres35) [reply apptools/preferences/preference_binding.py:124]: yeah I noticed this as well... not sure what the best thing to do is

**Comment by aaronayres35** [comment]
Orthogonal to this PR, but we should add a "maintenance" news fragment type. I wasn't sure which of "feature, bugfix, deprecation, removal, doc, test, build" to classify this PR as, but it feels like it will be ultimately relevant for the changelog (as I imagine 5.1 of apptools main purpose will be this migration to observe / depending on traits 6.2)

I am going to call it a feature for now simply because IDK where to put it. When generating the actual changelog we will want to move this, but this way it will still be picked up

**Pull request merged**